### PR TITLE
Encrypt password

### PR DESCRIPTION
**Pull request by andreiuta97**
Password will now appear encrypted in the database.

### CODE REVIEW

File: src/Entity/User.php [hunk 1]
```diff
@@ -93,7 +93,7 @@ public function setEmail(string $email)
      */
     public function setPassword(string $password)
     {
-        $this->password = $password;
+        $this->password = md5($password);
```
Comment (Raresmldvn) [inline src/Entity/User.php]: move this out of the setter. it is very counterintuitive, and you might end up md5-ing an md5

File: src/Controller/AuthenticationController.php [hunk 2]
```diff
@@ -55,7 +55,7 @@ public function redirectAtLogin(Request $request, array $requestAttributes): Res
             $body = Stream::createFromString('');
             $response = new Response($body, '1.1', 301, '');
             /** @var Response $redirect */
-            $redirect = $response->withHeader('Location', 'http://local.quiz.com/candidate/homepage?page=1');
+            $redirect = $response->withHeader('Location', 'http://local.quiz.com/candidate/homepage');
             return $redirect;
```
Comment (Raresmldvn) [inline src/Controller/AuthenticationController.php:69]: just return, no need to store it in another variable

File: src/Service/AuthenticationService.php [hunk 3]
```diff
@@ -44,15 +44,12 @@ public function getLoggedUser(): ?User
         if (!isset($_SESSION['id'])) {
             return null;
         }
```
Comment (Raresmldvn) [inline src/Service/AuthenticationService.php]: use the session service (inject it in this service), do not use globals directly.

File: src/Entity/User.php (hunk 1 above)
Comment (Raresmldvn) [reply src/Entity/User.php]: also, the entity should not care about the hashing. you should have a service that deals with password hashing.

File: src/Controller/AuthenticationController.php [hunk 4]
```diff
@@ -42,7 +42,7 @@ public function redirectAtLogin(Request $request, array $requestAttributes): Res
         $password = $request->getParameter('password');
         $role = $this->authenticationService->login($email, $password);
         $userRepo = $this->repositoryManager->getRepository(User::class);
-        $user = $userRepo->findOneBy(['email' => $email, 'password' => $password]);
+        $user = $userRepo->findOneBy(['email' => $email, 'password' => md5($password)]);
```
Comment (Raresmldvn) [inline src/Controller/AuthenticationController.php]: you should not place `md5` here. what if you want to change the hashing strategy? better make a `HashingService` (or how you want to call it) that deals with the hashing. 
Think about Strategy pattern and how at some point you would like to change the strategy in a future version

File: src/Controller/AuthenticationController.php (hunk 2 above)
Comment (andreiuta97) [reply src/Controller/AuthenticationController.php:69]: @Raresmldvn  All functions that redirect a response do it this way. I will make a separate issue for this one in which I want to make a function/class in mvc-framework to help me redirect responses

Comment (noonlit) [reply src/Controller/AuthenticationController.php:69]: he means that you can return $response->withHeader('Location', 'http://local.quiz.com/candidate/homepage') from the method, without storing the result in the $redirect variable first :)

File: src/Service/HashingService.php [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+
+namespace QuizApp\Service;
+
+
+class HashingService
```
Comment (noonlit) [inline src/Service/HashingService.php:7]: <3

File: src/Service/HashingService.php [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+{
+    public function hash(string $password): ?string
+    {
+        return password_hash($password, PASSWORD_BCRYPT);
```
Comment (Raresmldvn) [inline src/Service/HashingService.php]: pass that constant as an argument to the constructor of the service, in order to be able to easily change it. This is the lightest version of a Strategy pattern (though not really) :D

File: src/Service/AuthenticationService.php [hunk 7]
```diff
@@ -12,24 +12,31 @@ class AuthenticationService
 {
     private $userRepo;
     private $session;
+    private $hashingService;
```
Comment (Raresmldvn) [inline src/Service/AuthenticationService.php]: please place a newline between each pair of properties, looks better

**Pull request merged**